### ISSUE TITLE
SPA Core bundle does not start when installed on 6.5.0 - 6.5.7[bug]

### ISSUE THREAD

**Issue by jaydubb12** [adobe/aem-spa-project-core#27]
**Describe the bug**
When the react core package is installed directly or via Archetype 25, the core bundle does NOT start on v6.5.0, nor on v6.5.7. The error message is com.adobe.cq.wcm.core.components.models,version=[12.15,13) -- Cannot be resolved

**Root Cause**
* The core component version referenced in the pom.xml is 2.11.1 which has a version that does NOT meet the minimum requirements for the referenced package.

**Local env remediation**
*  Added core.components.all package as a subpackage to the all/pom.xml
*  Upgraded the core component version from 2.11.1 to 2.14 

**Recommended Fix**
*Upgrade the core components minimum version from 2.11.1 to the 2.14 release of the core components to resolve the issue
* Configure the bnd exports / bundle start priority configuration to export the package vs setting as a sub-package as noted in the all/pom.xml

**Related issue**
Once I upgraded the component version and deployed it locally, the bundles started, but then I noted a new issue was raised via the aem-analyze feature which states the following...which I will log on another ticket

[ERROR] com.adobe.aem:spa.project.core.core:1.2.3-SNAPSHOT: Bundle spa.project.core.core:1.2.3-SNAPSHOT is importing package(s) Package com.adobe.cq.wcm.core.components.models;version=[12.18,13) in start level 20 but no visible bundle is exporting these for that start level in the required version range.

**Package version**
1.2.0
1.2.3

**To Reproduce**
1. (Option 1) Create a project with the archetype properties referenced in the attachment, and install on a local author instance running 6.5.0 or 6.5.7

2. (Option 2) Install a clean instance of 6.5.0 or 6.5.7 and install the packages from release 1.2.0

**Expected behavior**
When packages are installed, the bundles will start

**Comment by pfauchere** [comment]
Internally tracked by CQ-4317712

**Comment by habansal** [comment]
### tldr;
Bundle starts fine when we would use correct parameter while generating a project from the Archetype.
Use `aemVersion=` `6.5.5` and NOT `cloud`

### Situation:
When trying to install the `aem-spa-project-core` bundle directly on a 6.5.7 instance, I could reproduce this issue.

### Investigation:
Basically, what happens is that the 6.5.0 sample-content-instance includes the `2.3.2` version of the core components.
Since, servicepacks don't include any core components package, 6.5.7 instance (sample-content) also has the `2.3.2` version of core components.

Now, The `2.3.2` core components export `12.7.0` version of `com.adobe.cq.wcm.core.components.models` package. See [1].
Whereas, `aem-spa-project-core-1.2.0` has a dependency (scope: provided) on `2.11.1` version of core components (See [2]) and hence expects `com.adobe.cq.wcm.core.components.models; version="[12.15, 13)`. See [3] or check manifest-headers of the `spa-project-core` bundle.

That is the reason the bundle doesn't activate.

### Feedback on the suggested fix:
Increasing the dependency to 2.14 would only further increase the version required for the `core.components.models` package. So, that does't help here and is not related to this issue.

Adding `core.components.all` package as a sub-package to the all/pom.xml might help, BUT, is not really needed.
Because i don't expect this `aem-spa-project-core` to be installed solo on the AEM instance. 
There would be some sample content like website or maybe a project generated from the archetype that would be installed on AEM.
The Archetype already, as expected, includes the Core components (for non-cloud AEM) and `spa-project-core`. See [4].

#### Verification that issue doesn't occur with Archetype 25:
Archetype 25 is already including the correct versions of Core components and spa-project-core -
`<spa.project.core.version>1.2.0</spa.project.core.version>`
and
`<core.wcm.components.version>2.13.0</core.wcm.components.version>`
See [5] and [6].

I also verified by generating a project from the archetype and installing on an AEM instance and all bundles are activated.
Then, i also tried re-installing `spa-project-core` on that instance and still all bundles were activated, as expected.

### Suspected mistake in reproduction steps reported - 
The gap here is that you are using `aemVersion=cloud` parameter while generating the archetype but installing that on a 6.5.7 instance. (Based on the file attached in https://github.com/adobe/aem-spa-project-core/issues/28)
You should use `aemVersion=6.5.7`.

_Please note -_
`Set aemVersion=cloud for AEM as a Cloud Service;
Set aemVersion=6.5.5 for Adobe Managed Services, or on-premise. The Core Components dependency is only added for non cloud aem versions as the Core Components are provided OOTB for AEM as a Cloud Service.`
Reference: 

### Conclusion:
Given that, I would suggest to close this issue and #28 as No fix is required here.

[1] : https://github.com/adobe/aem-core-wcm-components/blob/core.wcm.components.reactor-2.3.2/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/package-info.java#L37

[2] : https://github.com/adobe/aem-spa-project-core/blob/spa.project.core-1.2.0/pom.xml#L34

[3] : https://github.com/adobe/aem-core-wcm-components/blob/core.wcm.components.reactor-2.11.1/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/package-info.java#L37

[4] : https://experienceleague.adobe.com/docs/experience-manager-learn/spa-react-tutorial/create-project.html?lang=en#author-content (Build the project - point 6)

[5] : https://github.com/adobe/aem-project-archetype/blob/aem-project-archetype-25/src/main/archetype/pom.xml#L83

[6] : https://github.com/adobe/aem-project-archetype/blob/aem-project-archetype-25/src/main/archetype/pom.xml#L68

[7] : https://github.com/adobe/aem-project-archetype#usage
